agent/agent/src/main/java/com/microsoft/applicationinsights/agent/Main.java

```diff
@@ -0,0 +1,158 @@
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent;
+
+import static java.util.Arrays.asList;
+
+import com.microsoft.applicationinsights.agent.internal.Copy;
+import java.io.File;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.CodeSource;
+import java.util.List;
+import java.util.Map;
+import java.util.jar.Attributes;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.jar.JarOutputStream;
+import java.util.jar.Manifest;
+import java.util.regex.Pattern;
+
+@SuppressWarnings("SystemOut")
+public class Main {
+
+  private static final boolean DEBUG = Boolean.getBoolean("applicationinsights.debug.unsign");
+
+  private static final Pattern SIGNATURE_FILE = Pattern.compile("META-INF/.*\\.(RSA|SF)");
+
+  public static void main(String[] args) throws Exception {
+    if (args.length == 1 && args[0].equals("unsign")) {
+      unsign();
+    } else {
+      System.err.println("Incorrect usage");
+    }
+  }
+
+  private static void unsign() throws Exception {
+    Path agentJar = getAgentJar();
+
+    Path tempJar = Files.createTempFile("applicationinsights-", ".jar");
+
+    byte[] buffer = new byte[1024];
+
+    try (JarInputStream in = new JarInputStream(Files.newInputStream(agentJar))) {
+
+      Manifest inManifest = in.getManifest();
+
+      if (!isSigned(inManifest)) {
+        System.err.println();
+        System.err.println("The agent jar is already unsigned");
+        System.exit(0);
+      }
+
+      Manifest outManifest = new Manifest();
+      outManifest.getMainAttributes().putAll(inManifest.getMainAttributes());
+
+      JarOutputStream out = new JarOutputStream(Files.newOutputStream(tempJar), outManifest);
+
+      JarEntry entry;
+      while ((entry = in.getNextJarEntry()) != null) {
+        if (SIGNATURE_FILE.matcher(entry.getName()).matches()) {
+          continue;
+        }
+        out.putNextEntry(entry);
+        int read;
+        while ((read = in.read(buffer)) != -1) {
+          out.write(buffer, 0, read);
+        }
+      }
+
+      out.close();
```

Comment:
is out Autoclosed regardless?

Reply:
`out` wasn't used in try-with-resources, so wasn't, but changed it to use try-with-resources, which is better 👍